tket/src/Circuit/macro_manipulation.cpp

```diff
@@ -139,6 +139,9 @@ void Circuit::append_with_map(const Circuit& c2, const unit_map_t& qm) {
   Circuit copy = c2;
   copy.rename_units(qm);
 
+  copy.add_wasm_register(_number_of_wasm_wires);
+  add_wasm_register(copy._number_of_wasm_wires);
```

Comment:
What happens if the appended-to circuit already contains WASM wires? Will this throw an error?